### PR TITLE
UHM-5112: For Intake and Followup Forms, migrate Form Version

### PR DESCRIPTION
**Pull request by mogoodrich**

### CODE REVIEW

**Review by cioan — APPROVED** [review]
Thanks @mogoodrich for updating the get() function too!

**Review by lnball — COMMENTED** [review]
LGTM

File: configuration/messageproperties/messages_en.properties [hunk 1]
```diff
@@ -1280,6 +1280,8 @@ pihcore.visitNote.plan.medication.liver.disease=Medications for Liver Disease
 pihcore.visitNote.plan.medication.heart.failure=Medications for Heart Failure
 pihcore.ncd.htn.or.diabetes=Hypertension / Diabetes
 
+pihcore.hiv.legacyDataNotMigrated=LEGACY DATA from HIV EMR, not fully migrated
```
Comment (brandones) [inline configuration/messageproperties/messages_en.properties:1283]: Maybe we can elaborate on what this means for the end user?

```suggestion
pihcore.hiv.legacyDataNotMigrated=WARNING: This data comes from an old form in the old HIV EMR. Some data which may have been filled in on the old form might not appear here.
```

File: configuration/pih/scripts/visit/encounterTypeConfig.js [hunk 2]
```diff
@@ -1278,25 +1347,43 @@ angular.module("encounterTypeConfig", [])
...
+              if (encounterType[country].hasOwnProperty(site)) {
+                encounterType = encounterType[country][site];
+              } else {
+                encounterType = encounterType[country].hasOwnProperty('DEFAULT') ?
+                  encounterType[country]['DEFAULT'] : encounterType[country];
+              }
+            } else {
+              encounterType = encounterType.hasOwnProperty(['DEFAULT']) ?
+                encounterType['DEFAULT'] : encounterType;
+            }
```
Comment (brandones) [inline configuration/pih/scripts/visit/encounterTypeConfig.js:1373]: We can replace lines 1363-1373 with

```suggestion
              encounterType = encounterType[country][site] ||
                  encounterType[country]['DEFAULT'] ||
                  encounterType[country] ||
                  encounterType['DEFAULT'] ||
                  encounterType;
```

File: configuration/pih/scripts/visit/encounterTypeConfig.js [hunk 3]
```diff
@@ -1278,25 +1347,43 @@ angular.module("encounterTypeConfig", [])
...
+            // resolve any version-specific configs
+            if (encounterType.hasOwnProperty('versions')) {
+              if (encounter.form != null) {
+                encounterType = encounterType['versions'].hasOwnProperty(encounter.form.version) ?
+                  encounterType['versions'][encounter.form.version] : encounterType['versions']['DEFAULT']
+              }
+              else {
+                encounterType = encounterType['versions']['DEFAULT'];
+              }
             }
```
Comment (brandones) [inline configuration/pih/scripts/visit/encounterTypeConfig.js:1384]: Similarly,

```suggestion
            if (encounterType.versions) {
                encounterType = encounter.form && encounter.form.version ?
                    encounterType['versions'][encounter.form.version] || encounterType['versions']['DEFAULT'] :
                    encounterType['versions']['DEFAULT'];
            }
```

**Review by brandones — CHANGES_REQUESTED** [review]
A few requested changes, but generally looks good!

File: configuration/pih/scripts/visit/encounterTypeConfig.js (hunk 3 above)
Comment (brandones) [reply configuration/pih/scripts/visit/encounterTypeConfig.js:1384]: Actually, consider whether you want to `console.error` something helpful in the case that `encounter.form && !encounter.form.version`, assuming that would imply a programming error in the calling code.

File: configuration/pih/scripts/visit/encounterTypeConfig.js (hunk 2 above)
Comment (cioan) [reply configuration/pih/scripts/visit/encounterTypeConfig.js:1373]: Nice @brandones ! This is much cleaner and easier to read.

File: configuration/messageproperties/messages_en.properties (hunk 1 above)
Comment (mogoodrich) [reply configuration/messageproperties/messages_en.properties:1283]: Yeah, that was really just a starting message... wanted Dave and Debbie and others to take a look and determine what they really want it to say.  Happy to go with your approach as a first pass though...

File: configuration/pih/scripts/visit/encounterTypeConfig.js (hunk 2 above)
Comment (mogoodrich) [reply configuration/pih/scripts/visit/encounterTypeConfig.js:1373]: Thanks @brandones !

**Review by brandones — APPROVED** [review]

**Review by mseaton — APPROVED** [review]
Going to take your word on this PR.

**Pull request merged**